lighthouse-viewer/test/viewer-test-pptr.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* eslint-env mocha */
+
+const path = require('path');
+const assert = require('assert');
+const puppeteer = require('../../node_modules/puppeteer/index.js');
+
+const {server} = require('../../lighthouse-cli/test/fixtures/static-server.js');
+const portNumber = 10200;
+const viewerUrl = `http://localhost:${portNumber}/lighthouse-viewer/dist/index.html`;
+const sampleLhr = __dirname + '/../../lighthouse-core/test/results/sample_v2.json';
+
+const config = require(path.resolve(__dirname, '../../lighthouse-core/config/default-config.js'));
+const lighthouseCategories = Object.keys(config.categories);
+const getAuditsOfCategory = category => config.categories[category].audits;
+
+// TODO: should be combined in some way with lighthouse-extension/test/extension-test.js
+describe('Lighthouse Viewer', function() {
+  // eslint-disable-next-line no-console
+  console.log('\n✨ Be sure to have recently run this: yarn build-viewer');
+
+  let browser;
+  let viewerPage;
+  const pageErrors = [];
+
+  function getAuditElementsCount({category, selector}) {
+    return viewerPage.evaluate(
+      ({category, selector}) =>
+        document.querySelector(`#${category}`).parentNode.querySelectorAll(selector).length,
+      {category, selector}
+    );
+  }
+
+  before(async function() {
+    // eslint-disable-next-line
+    this.timeout(10 * 1000);
+
+    server.listen(portNumber, 'localhost');
+
+    // start puppeteer
+    browser = await puppeteer.launch({
+      headless: false,
+      executablePath: process.env.CHROME_PATH,
+      args: [],
+    });
+    viewerPage = await browser.newPage();
+    viewerPage.on('pageerror', pageError => pageErrors.push(pageError));
+    await viewerPage.goto(viewerUrl, {waitUntil: 'networkidle2'});
+    const fileInput = await viewerPage.$('#hidden-file-input');
+    await fileInput.uploadFile(sampleLhr);
+    await viewerPage.waitForSelector('.lh-container');
```

Comment:
Do you want to set a smaller timeout for this and goto? Default is 30s. If the link or gist fails, we should know about it sooner. Speed up the tests.

Reply:
whoops, yes, misread that as 3s and thought it sounded great :)